### PR TITLE
workflows: add external workload conformance test

### PR DESCRIPTION
**Pull request by nbusseneau**
This was ported and adapted from `cilium-cli`.

Supersedes #16382.

### CODE REVIEW

**Comment by nbusseneau** [comment]
The changes in this PR work using the exact same commits on top of `v1.10.3`: https://github.com/cilium/cilium/actions/runs/1050340820 (failure is due to flow validation not working, but I expected that. It did run the connectivity test, no VM <-> cluster connectivity issue).

This means we have a regression in `master` with external workloads.

**Comment by nbusseneau** [comment]
Regression comes from commit 0681343309ef15677c9335802bd724500f1d663d (PR #15632):
- Failing workflow run, right on top of this commit: https://github.com/cilium/cilium/actions/runs/1059788504
- Successful workflow run on previous commit in `master`: https://github.com/cilium/cilium/actions/runs/1059727108
- Successful workflow run on current `master` with the commit reverted: https://github.com/cilium/cilium/actions/runs/1059836909

**Comment by nbusseneau** [comment]
Opened #16984 to revert regression commit, this PR is now blocked until #16984 is merged.

**Comment by nbusseneau** [comment]
Following discussion in #16984, we found the issue and opened a fix at #16986. This PR is now blocked until #16986 is merged.

**Comment by nbusseneau** [comment]
#16986 is merged, rebased and marked as ready for review.

**Comment by nbusseneau** [comment]
Link to test run of workflow changes: https://github.com/cilium/cilium/actions/runs/1074746971
No need to run rest of CI since it's a new workflow. Removing temporary test commit.

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (pchaigno) [inline .github/workflows/conformance-externalworkloads-v1.10.yml:156]: Not specific to this PR, but why don't we set the commit status before checking out the code? I see the dependency of this step on the `vars` step, but couldn't we move both of these to the beginning of the job?

**Review by nebril — APPROVED** [review]

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (pchaigno) [inline .github/workflows/conformance-externalworkloads-v1.10.yml:1]: What's this file about?

Comment (nbusseneau) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:1]: Testing for backport branches via `test-backport-1.10`, see #16944 :)

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (nbusseneau) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:156]: Sure, could be done, I did not do it mostly because it was more convenient to keep everything in a single `vars` job. Actually, before we added stuff around, it was the first thing in the workflow :)

Checking out the code is almost instantaneous though (5s) so it does not really matter.

Comment (pchaigno) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:156]: Why would that prevent us from keeping everything in a single `vars` step?

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (pchaigno) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:1]: :scream: :scream: :scream:

Comment (pchaigno) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:1]: Is it too late to ask to go back to Jenkins?

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (nbusseneau) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:156]: I was thinking you were saying to put the job step above everything else (i.e. before `Deduce required tests from code changes`) but re-reading it seems you were only suggesting to swap the order of the steps in the second job. Will do.

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (nbusseneau) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:1]: I'll be working on removing that horrible thing ASAP 😅

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (nbusseneau) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:156]: Done :)

Comment (pchaigno) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:156]: :rocket: I can send a PR to do the same in other workflows. It's only a couple seconds, but the queuing time is also short in general (when load < 1), so a couple seconds could make a visible difference.

**Review by pchaigno — APPROVED** [review]

File: .github/workflows/conformance-externalworkloads-v1.10.yml
Comment (nbusseneau) [reply .github/workflows/conformance-externalworkloads-v1.10.yml:156]: I don't think we have it in other workflows 🤔

**Pull request merged**
